src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(...args) {
+    if (args.length === 0) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += args.reduce((acc, value) => acc + value, 0);
```

Comment:
The use of `reduce` is unnecessary here since the `adder` function is meant to handle only one argument at a time. Instead, you should directly add the single argument to the `sum` variable, if it is not undefined.